test/react-native/features/fixtures/scenario-launcher/lib/ScenarioLauncher.js

```diff
@@ -39,6 +36,85 @@ export const launchScenario = async (rootTag) => {
     appParams.initialProps = { concurrentRoot: true }
   }
 
-  AppRegistry.registerComponent(command.scenario_name, () => scenario.App)
-  AppRegistry.runApplication(command.scenario_name, appParams)
+  AppRegistry.registerComponent(scenarioName, () => scenario.App)
+  AppRegistry.runApplication(scenarioName, appParams)
+}
+
+async function writePersistedStateFile (contents) {
+  if (!await FileSystem.exists(PERSISTED_STATE_DIRECTORY)) {
+    console.error(`[BugsnagPerformance] creating persisted state directory: ${PERSISTED_STATE_DIRECTORY}`)
+    await FileSystem.mkdir(PERSISTED_STATE_DIRECTORY)
+  }
+
+  console.error(`[BugsnagPerformance] writing to: ${PERSISTED_STATE_PATH}`)
+
+  await FileSystem.writeFile(
+    PERSISTED_STATE_PATH,
+    JSON.stringify(contents)
+  )
```

Comment:
does `writeFile` append the contents here? or would calling `setSamplingProbability` overwrite the data from `setDeviceId` and vice versa?

Reply:
There's no merging of the contents so if you called both the second would win